### PR TITLE
Supported option to specify site name for multi-site deployments

### LINKED ISSUE #26460: Supported option to specify site name for multi-site deployments
### Description

With multi-site deployments each site has to have a unique name for the embedded Infinispan.

This is currently set with the unsupported `jboss.site.name` system property. See https://www.keycloak.org/high-availability/connect-keycloak-to-external-infinispan.

A supported option `cache-site-name` should be introduced for this purpose. This could be done just as an SPI config option and exposed at https://www.keycloak.org/server/all-provider-config.

### Discussion

_No response_

### Motivation

_No response_

### Details

_No response_

### PR DESCRIPTION
**Pull request by vmuzikar**
Closes #26460

### CODE REVIEW

File: quarkus/tests/integration/src/test/java/org/keycloak/it/storage/database/ExternalInfinispanTest.java [hunk 1]
```diff
@@ -28,10 +33,16 @@
...
+@TestMethodOrder(MethodOrderer.OrderAnnotation.class)
 public class ExternalInfinispanTest {
 
     @Test
-    @Launch({ "start-dev", "--features=multi-site", "--cache=ispn", "--cache-config-file=../../../test-classes/ExternalInfinispan/kcb-infinispan-cache-remote-store-config.xml", "-Djboss.site.name=ISPN" })
+    @Launch({ "start-dev", "--features=multi-site", "--cache=ispn", "--cache-config-file=../../../test-classes/ExternalInfinispan/kcb-infinispan-cache-remote-store-config.xml", "--spi-connections-infinispan-quarkus-site-name=ISPN" })
+    // Needs to be executed as last; removing caches in this messes up with cache manager clean up after the test causing the following exception upon next start (possible ISPN bug?):
+    // org.infinispan.client.hotrod.exceptions.HotRodClientException:: ISPN004091: MBean registration failed
+    // ...
+    // Caused by: javax.management.InstanceAlreadyExistsException: org.infinispan:type=HotRodClient,name=Default,cache=sessions
```
Comment (vmuzikar) [inline quarkus/tests/integration/src/test/java/org/keycloak/it/storage/database/ExternalInfinispanTest.java:44]: @keycloak/store Can you please review as well? I'm not sure particularly about this if it's a bug, or expected behaviour.

Comment (ahus1) [reply quarkus/tests/integration/src/test/java/org/keycloak/it/storage/database/ExternalInfinispanTest.java]: Hi @vmuzikar - I've added something which I think is a workaround. 

I've raised an upstream issue. This should be good to be merged now.

**Review by ahus1 — DISMISSED** [review]
Reviewed and tested it. Thank you for this change - the SPI parameter should be good enough for the HA guide.

**Review by mabartos — DISMISSED** [review]
LGTM

**Comment by vmuzikar** [comment]
@ahus1 Thanks for the review and the changes! I noticed there was accidentally missing `@Test` annotation from `testSiteNameAsSystemProperty` after the changes so the test didn't run and we could not be sure the workaround was no longer needed. I re-added that in a separate commit.

**Review by ahus1 — APPROVED** [review]

**Comment by ahus1** [comment]
> I noticed there was accidentally missing `@Test` annotation from `testSiteNameAsSystemProperty`

Oops! Thanks for spotting this, @vmuzikar!

**Pull request merged**